paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -330,8 +318,10 @@ internal class PaymentSheetViewModel @Inject internal constructor(
                 deferredIntentConfirmationType = pendingResult.deferredIntentConfirmationType,
                 finishImmediately = true
             )
+        } else if (state.validationError != null) {
+            handlePaymentSheetStateLoadFailure(state.validationError)
         } else {
-            handlePaymentSheetStateLoadFailure(error)
+            initializeWithState(state)
         }
```

Comment:
The source of truth here moves from the loaded intent to the handler deciding whether or not we should instantiate the `PaymentSheet` state or not.